maintainers_annotation/maintainers_annotation.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package maintainers_annotation
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/checks/fileparser"
+	"github.com/ossf/scorecard/v4/clients"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"gopkg.in/yaml.v3"
+)
+
+var errNoScorecardYmlFile = errors.New("scorecard.yml doesn't exist or file doesn't contain exemptions")
+
+// AnnotationReason is the reason behind an annotation.
+type AnnotationReason string
+
+const (
+	// AnnotationTestData is to annotate when a check or probe is targeting a danger
+	// in files or code snippets only used for test or example purposes.
+	AnnotationTestData AnnotationReason = "test-data"
```

Comment:
nit/style:

There's a lot of repetition in the variable names. Callers would access:
```go
maintainers_annotation.AnnotationTestData
```

Recommend 
```go
type Reason string

const (
    TestData Reason = "test-data"
    // etc
```

so it reads better:
```go
maintainers_annotation.TestData
```

Reply:
Aw, sure. I was sort of copying a pattern from `checks` like `checks.CheckBinaryArtifacts`, but I agree it causes too much repetition, noted.

Reply:
Done.